### PR TITLE
Adds a roadmap and versioning documentation

### PR DESCRIPTION
**Pull request by Zanadar**
This PR is a place to discuss documentation in support of our CNCF incubation; specifically:
- Our roadmap
~~- governance~~ (covered by #2)
- Our versioning scheme

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Contributing to the Cloud Native Buildpacks (CNB) project
```
Comment (Zanadar) [inline CONTRIBUTING.md]: To be filled out soon.

File: GOVERNANCE.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+CNB Governance
```
[the remark this answers is not in the document]
Comment (Zanadar) [reply GOVERNANCE.md]: Needs some formatting.

[the remark this answers is not in the document]
Comment (Zanadar) [reply GOVERNANCE.md]: Will remove in favor #2

File: VERSIONING.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#### Versioning in the CNB project
+The CNB project has a variety of versioning schemes that contribute to overall stability and communicates the compatibility of different components.
+
+#### Lifecycle
```
Comment (Zanadar) [inline VERSIONING.md]: ### Spec versioning

File: VERSIONING.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    - When `<major>` is `0`, the platform is only compatible with lifecycles implementing that exact Platform API.
+    - When `<major>` is greater than `0`, the platforms is only compatible with lifecycles implementing platform API
+    `<major>.<minor>`, where `<major>` of the lifecycle equals
+
+### Buildpack API Version
+Documented in the [buildpack spec](https://github.com/buildpacks/spec/blob/master/buildpack.md#buildpacktoml-toml) this field indicate compatibility with a given lifecycle according to the following rules:
+- When <major> is 0, the buildpack is only compatible with lifecycles implementing that exact buildpack API.
+- When <major> is greater than 0, the buildpack is only compatible with lifecycles implementing buildpack API <major>.<minor>, where <major> of the lifecycle equals <major> of the buildpack and <minor> of the lifecycle is greater than or equal to <minor> of the buildpack.
+
+#### Releases
```
Comment (Zanadar) [inline VERSIONING.md]: ### Implementations

File: ROADMAP.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
+# Roadmap
```
Comment (hone) [inline ROADMAP.md]: It's weird to have two Readmap headers?

Comment (hone) [reply ROADMAP.md]: Oh, I see. Can we move this into that `ROADMAP.md` for now?

File: VERSIONING.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# Versioning in the CNB project
+The CNB project versions our [spec components](#spec) and versions our [implementations](#implementations) following seperate schemas, detailed below:
```
Comment (hone) [inline VERSIONING.md]: Do we need to document how libbuildpack is versioned too?

**Comment by hone** [comment]
This should also link out to our [Code of Conduct](https://github.com/buildpacks/.github/blob/master/CODE_OF_CONDUCT.md).

File: ROADMAP.md (hunk 5 above)
Comment (Zanadar) [reply ROADMAP.md]: Its in at file already as far as I can tell.

File: VERSIONING.md (hunk 6 above)
Comment (Zanadar) [reply VERSIONING.md]: Good call will do.

File: VERSIONING.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
```
Comment (jkutner) [inline VERSIONING.md]:
```suggestion
The CNB project versions our [spec components](#spec) and [implementations](#implementations) following seperate schemas, detailed below:
```

File: VERSIONING.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+
+### Spec component versioning
+[spec]: #spec
+Currently the spec describes three components which are seperately versioned. These APIs are versioned as follows:
```
Comment (jkutner) [inline VERSIONING.md]:
```suggestion
The spec describes three components that are separately versioned. These APIs are versioned as follows:
```

File: VERSIONING.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    - When `<major>` is greater than `0`, the platforms is only compatible with lifecycles implementing platform API
+    `<major>.<minor>`, where `<major>` of the lifecycle equals
+
+#### Buildpack API Version
+Documented in the [buildpack spec](https://github.com/buildpacks/spec/blob/master/buildpack.md#buildpacktoml-toml) this field indicate compatibility with a given lifecycle according to the following rules:
+- When <major> is 0, the buildpack is only compatible with lifecycles implementing that exact buildpack API.
+- When <major> is greater than 0, the buildpack is only compatible with lifecycles implementing buildpack API <major>.<minor>, where <major> of the lifecycle equals  <major> of the buildpack and <minor> of the lifecycle is greater than or equal to <minor> of the buildpack.
+
+#### Distribution API Version
+This component is currently not explicely versioned. As changes to this spec occur, an RFC will codify the contract.
```
Comment (jkutner) [inline VERSIONING.md]:
```suggestion
This component is not explicitly versioned. As changes to this spec occur, an codification of the contract may be proposed in a future RFC.
```

File: VERSIONING.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+
+### Implementation versioning
+[implementations]: #implementations
+
+The CNB projects contains two main implementations of the spec are versioning as follows:
```
Comment (jkutner) [inline VERSIONING.md]: I'm not sure what this is trying to say

Comment (Zanadar) [reply VERSIONING.md]: Me neither. Clarified the language a bit.

**Comment by Zanadar** [comment]
@jkutner @hone LMK what you think

File: VERSIONING.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    - When `<major>` is greater than `0`, the platforms is only compatible with lifecycles implementing platform API
+    `<major>.<minor>`, where `<major>` of the lifecycle equals
+
+#### Buildpack API Version
+Documented in the [buildpack spec](https://github.com/buildpacks/spec/blob/master/buildpack.md#buildpacktoml-toml) this field indicate compatibility with a given lifecycle according to the following rules:
+- When <major> is 0, the buildpack is only compatible with lifecycles implementing that exact buildpack API.
+- When <major> is greater than 0, the buildpack is only compatible with lifecycles implementing buildpack API <major>.<minor>, where <major> of the lifecycle equals  <major> of the buildpack and <minor> of the lifecycle is greater than or equal to <minor> of the buildpack.
+
+#### Distribution API Version
+This component is not explicitly versioned. As changes to this spec occur, an codification of the contract may be proposed in a future RFC.
```
Comment (Zanadar) [inline VERSIONING.md]: Should be:
> A codification

File: VERSIONING.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+[spec]: #spec
+The spec describes three components that are separately versioned. These APIs are versioned as follows:
+
+#### Platform API Version
+Given by CNB platforms to the lifecycle, this version indicates which platform API versions that platform currently support. In review, in [PR#67](https://github.com/buildpacks/spec/pull/67/files), this version indicates the compatibility between a CNB platform and a given lifecycle according to the following rules:
+    - When `<major>` is `0`, the platform is only compatible with lifecycles implementing that exact Platform API.
+    - When `<major>` is greater than `0`, the platforms is only compatible with lifecycles implementing platform API
+    `<major>.<minor>`, where `<major>` of the lifecycle equals
+
+#### Buildpack API Version
```
Comment (hone) [inline VERSIONING.md]: https://github.com/buildpacks/rfcs/blob/master/text/0010-api-versions.md

**Comment by Zanadar** [comment]
:shipit: !!

**Pull request merged**